File: src/ManaFoodProduct.Application/UseCases/ProductUseCase/Commands/UpdateProduct/UpdateProductCommand.cs
using MediatR;
using ManaFoodProduct.Application.Dtos;
using System.Collections.Generic;

namespace ManaFoodProduct.Application.UseCases.ProductUseCase.Commands.UpdateProduct
{
    public record UpdateProductCommand(Guid Id, string Name, double UnitPrice, Guid CategoryId, List<Guid>? ItemIds = null) : IRequest<ProductDto>;
}

File: src/main/kotlin/br/com/manafood/manafoodproduct/application/usecase/product/commands/update/UpdateProductCommand.kt
package br.com.manafood.manafoodproduct.application.usecase.product.commands.update

import java.util.*

data class UpdateProductCommand(
    val id: UUID,
    val name: String,
    val description: String?,
    val unitPrice: Double,
    val itemIds: List<UUID> = emptyList(),
    val categoryId: UUID,
    val updatedBy: UUID
)
